packages/integration-karma/test/component/decorators/api/index.spec.js

```diff
@@ -111,3 +111,17 @@ it('should not log an error when initializing api value to null', () => {
 
     expect(() => document.body.appendChild(elm)).not.toLogErrorDev();
 });
+
+describe('restrictions', () => {
+    it('throws a property error when a public field conflicts with a method', () => {
+        expect(() => {
+            // eslint-disable-next-line @typescript-eslint/no-unused-vars
+            class Invalid extends LightningElement {
```

Comment:
It is not intuitive from the testcase how the component class is processed by engine-core. I assume when we rollup the test suite for execution, the component class invokes registerDecorators and it automatically triggers the validations. A comment explaining that would be helpful.